engine/api/application_vulnerability.go

```diff
@@ -28,25 +29,30 @@ func (api *API) postVulnerabilityHandler() service.Handler {
 			return sdk.WrapError(err, "Unable to read body")
 		}
 
-		app, errL := application.LoadByName(api.mustDB(), key, appName)
-		if errL != nil {
-			return sdk.WrapError(errL, "Unable to load application")
+		proj, err := project.Load(api.mustDB(), projectKey)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load project %s", projectKey)
 		}
 
-		vulnDB, errV := application.LoadVulnerability(api.mustDB(), app.ID, vulnID)
-		if errV != nil {
-			return sdk.WrapError(errV, "Unable to load vulnerability")
+		app, err := application.LoadByProjectIDAndName(ctx, api.mustDB(), proj.ID, appName)
```

Comment:
If we just need the application ID, we shouldn't have to load the whole application because appName is an identifier.

Reply:
We first load the app to be able to check if it exists.

Reply:
We first load the app to be able to check if it exists.